BostonData/lambda_function/trash_intent.py

```diff
@@ -52,22 +54,119 @@ def get_trash_day_info(intent, session):
+    """
+    Determines the trash and recycling days for the provided address
+
+    :param address: String of address to find trash day for
+    :return: arrays containing next trash and recycling days
+    """
+
+    api_params = get_address_api_info(address)
+    if not api_params:
+        raise InvalidAddressError
+
+    trash_data = get_trash_day_data(api_params)
+    if not trash_data:
+        raise BadAPIResponse
+
+    trash_days, recycling_days = get_trash_days_from_trash_data(trash_data)
+
+    return trash_days, recycling_days
+
+
+def get_address_api_info(address):
+    """
+    Gets the parameters required for the ReCollect API call
+
+    :param address: Address to get parameters for
+    :return: JSON object containing API parameters with format:
+
+    {
+        'area_name': value,
+        'parcel_id': value,
+        'service_id': value,
+        'place_id': value,
+        'area_id': value,
+        'name': value
+    }
+
+    """
+
+    base_url = "https://recollect.net/api/areas/" \
+               "Boston/services/310/address-suggest"
+    url_params = {'q': address, 'locale': 'en-US'}
+    request_result = requests.get(base_url, url_params)
+
+    if request_result.status_code != requests.codes.ok:
+        print("Error getting ReCollect API info. Got response: {}"
+              .format(request_result.status_code))
+        return {}
+
+    if not request_result.json():
+        return {}
+
+    return request_result.json()[0]
+
+
+def get_trash_day_data(api_parameters):
+    """
+    Gets the trash day data from ReCollect using the provided API parameters
+
+    :param api_parameters: Parameters for ReCollect API
+    :return: JSON object containing all trash data
+    """
+
+    # Rename the default API parameter "name" to "formatted_address"
+    if "name" in api_parameters:
+        api_parameters["formatted_address"] = api_parameters.pop("name")
+
+    base_url = "https://recollect.net/api/places"
+    request_result = requests.get(base_url, api_parameters)
+
+    if request_result.status_code != requests.codes.ok:
+        print("Error getting trash info from ReCollect API info. "
+              "Got response: {}".format(request_result.status_code))
+        return {}
+
+    return request_result.json()
+
+
+def get_trash_days_from_trash_data(trash_data):
     """
-    Converts the string of day initials in the JSON response to the trash
-    day calendar API to a list of days.
+    Parse trash data from ReCollect service and return the trash and recycling
+    days.
+
+    :param trash_data: Trash data provided from ReCollect API
+    :return: Two arrays, with trash days and recycling days
+    """
+
+    try:
+        trash_days_string = trash_data["next_event"]["zone"]["title"]
+        trash_days = trash_days_string.replace('&', '').split()
+    except KeyError:
+        # ReCollect API returned an unexpected JSON format
+        raise BadAPIResponse
+
+    return trash_days, trash_days

```

Comment:
If we are not getting an explicit value back for recycling days, I think we should just leave out recycling for now and adjust our response script to that effect rather than assuming trash/recycling are on the same day (unless we can confirm that to be true... but either way we can just return the one value and adjust the response script).